linkcheck/checker/proxysupport.py

```diff
@@ -17,12 +17,14 @@
 """
 Mixin class for URLs that can be fetched over a proxy.
 """
-import urllib
-try:
-    import urlparse
-except ImportError:
-    # Python 3
+try: # Python 3
     from urllib import parse as urlparse
+    from urllib import request as urlrequest
+    from urllib.parse import splitport as urllib_splitport
+except ImportError:
+    from urllib import splitport as urllib_splitport
+    import urllib as urlrequest
+    import urlparse
```

Comment:
i feel this looks a little off... in the long term, shouldn't we be looking at using Python3 semantics? e.g. we would use `urllib.parse` instead of `urlparse`...